### PR TITLE
Doc linklint fixups

### PR DESCRIPTION
**Pull request by bpkroth**
# Pull Request

## Title

Doc linklint fixups

---

## Description

GithubPages serves our docs out of a `/MLOS` path, but previously we checked things out of `/` using an Nginx docker container.

This PR changes the Nginx docker container used for testing to also serve paths out of `/MLOS` and fixes the broken links found from that.

- [x] Adjust Nginx to serve from `/MLOS` like Github pages.
- [x] Fix linklint errors found from that
- [x] spelling

- See #891 

---

## Type of Change

- 📝 Documentation update

---

## Testing

- Existing CI (which includes docstring and some link lint checks)

---

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ MLOS Documentation
 .. image:: badges/coverage.svg
    :target: htmlcov/index.html
 
-`MLOS <source_tree_docs/index.html>`_ is a project to enable `autotuning <autoapi/mlos_core/index.html>`_ with `mlos_core <autoapi/mlos_core/index.html>`_ for systems via `automated benchmarking <autoapi/mlos_bench/index.html>`_ with `mlos_bench </autoapi/mlos_bench/index.html>`_ including managing the storage and `visualization <autoapi/mlos_viz/index.html>`_ of the results via `mlos_viz <autoapi/mlos_viz/index.html>`_.
+`MLOS <source_tree_docs/index.html>`_ is a project to enable `autotuning <autoapi/mlos_core/index.html>`_ with `mlos_core <autoapi/mlos_core/index.html>`_ for systems via `automated benchmarking <autoapi/mlos_bench/index.html>`_ with `mlos_bench <autoapi/mlos_bench/index.html>`_ including managing the storage and `visualization <autoapi/mlos_viz/index.html>`_ of the results via `mlos_viz <autoapi/mlos_viz/index.html>`_.
```
Comment (bpkroth) [inline doc/source/index.rst:9]: Link error when deployed to GitHub Pages.

File: doc/nginx-default.conf [hunk 2]
```diff
@@ -5,10 +5,12 @@ server {
 
     #access_log  /var/log/nginx/host.access.log  main;
 
+    absolute_redirect off;
+    rewrite ^/$ /MLOS/ redirect;
```
Comment (bpkroth) [inline doc/nginx-default.conf:9]: Convenience to redirect from `http://localhost:8080` to `http://localhost:8080/MLOS` but only for the top level path.

File: doc/nginx-default.conf [hunk 3]
```diff
@@ -5,10 +5,12 @@ server {
...
+
     # Expects ./doc to be mapped to /doc in the nginx container.
-    location / {
-        #root   /usr/share/nginx/html;
-        root    /doc/build/html/;
+    location /MLOS/ {
+        alias   /doc/build/html/;
```
Comment (bpkroth) [inline doc/nginx-default.conf:13]: Adjust the base path for everything else to be more like Github Pages.

**Review by motus — APPROVED** [review]
Looks good!

**Pull request merged**